src/main/java/org/junitpioneer/jupiter/ReportEntryExtension.java

```diff
@@ -16,33 +16,78 @@
 import static org.junitpioneer.jupiter.ReportEntry.PublishCondition.ON_FAILURE;
 import static org.junitpioneer.jupiter.ReportEntry.PublishCondition.ON_SUCCESS;
 
+import java.lang.reflect.Method;
 import java.util.Arrays;
+import java.util.List;
 import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import java.util.stream.Stream;
 
 import org.junit.jupiter.api.extension.BeforeEachCallback;
 import org.junit.jupiter.api.extension.ExtensionConfigurationException;
 import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.InvocationInterceptor;
+import org.junit.jupiter.api.extension.ReflectiveInvocationContext;
 import org.junit.jupiter.api.extension.TestWatcher;
 
-class ReportEntryExtension implements TestWatcher, BeforeEachCallback {
+class ReportEntryExtension implements TestWatcher, BeforeEachCallback, InvocationInterceptor {
+
+	private static final ExtensionContext.Namespace NAMESPACE = ExtensionContext.Namespace
+			.create(ReportEntryExtension.class);
+
+	private static final String KEY = "ReportEntry";
 
 	@Override
 	public void beforeEach(ExtensionContext context) {
-		findAnnotations(context).forEach(ReportEntryExtension::verifyKeyValueAreNotBlank);
+		findAnnotations(context).forEach(entry -> verifyReportEntry(context, entry));
 	}
 
 	private Stream<ReportEntry> findAnnotations(ExtensionContext context) {
 		return PioneerAnnotationUtils.findAllEnclosingRepeatableAnnotations(context, ReportEntry.class);
 	}
 
+	private static void verifyReportEntry(ExtensionContext context, ReportEntry entry) {
+		verifyParameterCount(context, entry);
+		verifyKeyValueAreNotBlank(entry);
+		verifyKeyNotParameterized(entry);
+	}
+
+	private static void verifyParameterCount(ExtensionContext context, ReportEntry entry) {
+		boolean valueReferencesNumberedParameter = entry.value().matches(".*\\{[0-9]+}.*");
```

Comment:
Maybe extract `matches(".*\\{[0-9]+}.*")` into a method?

Reply:
Agree, extracted and slightly refactored.